tailcfg/tailcfg.go

```diff
@@ -927,13 +927,24 @@ func (hi *Hostinfo) TailscaleSSHEnabled() bool {
 
 func (v HostinfoView) TailscaleSSHEnabled() bool { return v.ж.TailscaleSSHEnabled() }
 
-// TailscaleFunnelEnabled reports whether or not this node has explicitly
-// enabled Funnel.
+// TailscaleFunnelEnabled reports whether or not this node has explicitly enabled Funnel.
 func (hi *Hostinfo) TailscaleFunnelEnabled() bool {
+	// TODO(irbekrm): once control uses TailscaleFunnelUsageDetected to determine whether funnel is enabled for
+	// clients with capver <= 112, use hi.IngressEnabled here.
 	return hi != nil && hi.WireIngress
 }
 
-func (v HostinfoView) TailscaleFunnelEnabled() bool { return v.ж.TailscaleFunnelEnabled() }
+// TailscaleFunnelUsageDetected reports whether the node has any funnel config applied to it, including one which
+// contains only disabled endpoints.
+func (hi *Hostinfo) TailscaleFunnelUsageDetected() bool {
```

Comment:
Alternatively, we could have a single method that accepts capver, so that the caller does not need to do the `if capver >= 113 then`. But that does not seem to fit the pattern used.
Or we could send capver in Hostinfo, but then that would be more data sent